### PR TITLE
vulkan-* 1.3.289

### PR DESCRIPTION
**Pull request by pthariensflame**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

### CODE REVIEW

**Comment by chenrui333** [comment]
looks like just vulkan-tools missing 1.3.281 release?

**Comment by pthariensflame** [comment]
> looks like just vulkan-tools missing 1.3.281 release?

That's correct!  Should we just revision-bump it for now?

**Comment by cho-m** [comment]
> > looks like just vulkan-tools missing 1.3.281 release?
> 
> That's correct! Should we just revision-bump it for now?

It sounds like upstream is skipping 281 - https://github.com/KhronosGroup/Vulkan-Tools/pull/975#issuecomment-2018697086

**Comment by pthariensflame** [comment]
@cho-m @chenrui333 Should we just skip this one, then, and only do 1.3.282 all at once?

**Comment by chenrui333** [comment]
sounds good to me.

**Comment by pthariensflame** [comment]
@chenrui333 Should be ready!

**Comment by pthariensflame** [comment]
I have no idea what went wrong in CI.  Getting permission errors while installing a normal bottle to the expected prefix shouldn’t happen.

**Comment by cho-m** [comment]
`vulkan-tools` needs some updates https://github.com/KhronosGroup/Vulkan-Tools/commit/c5d71df31f3e215d114c894046df8380c9403e02:
```
  Error: inreplace failed
  /private/tmp/vulkan-tools-20240423-14307-mq1pf7/Vulkan-Tools-1.3.283/cube/macOS/cube/CMakeLists.txt:
    expected replacement of "${MOLTENVK_DIR}/MoltenVK/dylib/macOS/libMoltenVK.dylib" with "${MOLTENVK_DIR}/lib/libMoltenVK.dylib"
  /private/tmp/vulkan-tools-20240423-14307-mq1pf7/Vulkan-Tools-1.3.283/cube/macOS/cubepp/CMakeLists.txt:
    expected replacement of "${MOLTENVK_DIR}/MoltenVK/dylib/macOS/libMoltenVK.dylib" with "${MOLTENVK_DIR}/lib/libMoltenVK.dylib"
```

**Comment by pthariensflame** [comment]
> VK_ERROR_INCOMPATIBLE_DRIVER: Vulkan is not supported on this device. MoltenVK requires Metal, which is not available on this device.

Okay.  I guess *all* the macOS VMs have this problem now.  :/

**Comment by pthariensflame** [comment]
> `vulkan-tools` needs some updates [KhronosGroup/Vulkan-Tools@c5d71df](https://github.com/KhronosGroup/Vulkan-Tools/commit/c5d71df31f3e215d114c894046df8380c9403e02):
> 
> ```
>   Error: inreplace failed
>   /private/tmp/vulkan-tools-20240423-14307-mq1pf7/Vulkan-Tools-1.3.283/cube/macOS/cube/CMakeLists.txt:
>     expected replacement of "${MOLTENVK_DIR}/MoltenVK/dylib/macOS/libMoltenVK.dylib" with "${MOLTENVK_DIR}/lib/libMoltenVK.dylib"
>   /private/tmp/vulkan-tools-20240423-14307-mq1pf7/Vulkan-Tools-1.3.283/cube/macOS/cubepp/CMakeLists.txt:
>     expected replacement of "${MOLTENVK_DIR}/MoltenVK/dylib/macOS/libMoltenVK.dylib" with "${MOLTENVK_DIR}/lib/libMoltenVK.dylib"
> ```

And given the root of this problem, we may be blocked until we upgrade `molten-vk`.

**Comment by pthariensflame** [comment]
Trying things by reincorporating #166005 before the rest of the commits.

**Comment by pthariensflame** [comment]
Yeah.  Between the macOS 13 runners having weird DB read issues, and AS Mac runners starting to exhibit the same missing-Metal problem previously plaguing Intel Mac runners, I'm inclined to think this is a CI issue.  Who should we ask to take a look at that?

**Comment by cho-m** [comment]
> the macOS 13 runners having weird DB read issues

Wasn't the workaround was added in https://github.com/Homebrew/homebrew-core/pull/149337 to deal with this?

**Comment by pthariensflame** [comment]
Okay; I've readded the thing from #149337; now macOS 13 passes!  But macOS 14 and macOS 12 don't, for different apparent reasons each.  macOS 12 runners have the missing-Metal issue; macOS 14 runners have some file conflict with `spirvCross.a` that prevented `molten-vk` from building.

**Comment by pthariensflame** [comment]
And this time the macOS 12 Arm runner succeeded?????

**Comment by ottob** [comment]
moltenvk 1.2.9 has been released now. maybe that will work better?

https://github.com/KhronosGroup/MoltenVK/releases/tag/v1.2.9

**Comment by pthariensflame** [comment]
Just waiting on `vulkan-extensionlayers` again now.

**Comment by pthariensflame** [comment]
@cho-m @chenrui333 @ottob Even though Vulkan 1.3.285 is starting to exist (`vulkan-headers` has already been updated`), I'm choosing to stick with 1.3.284 for this PR, assuming everything builds okay now, just so we don't have to delay any further.  I'll produce the 1.3.285 PR once this one is merged (if that's soon enough).

**Comment by pthariensflame** [comment]
Okay, no, MoltenVK is still failing to build for some reason.

**Comment by pthariensflame** [comment]
I guess we'll see if 1.3.285 makes any difference.

**Comment by chenrui333** [comment]
@pthariensflame can you maybe create a new PR, I generally dont recommend it, but it would help with attentions and also see how things flow differently (also looks like we can move vulkan-volk into separate PR if possible).

**Comment by pthariensflame** [comment]
> @pthariensflame can you maybe create a new PR, I generally dont recommend it, but it would help with attentions and also see how things flow differently (also looks like we can move vulkan-volk into separate PR if possible).

I am, unfortunately, going to be away from my computer for a month starting tonight!  (I have my phone still but I’m not trusting myself to do git actions unchecked from there.)  I can do as you ask when I’m back, updating to inevitably more recent versions of everything in the process, or you or another person can take things over for that time.  It’s up to you!

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
```
  ==> Testing vulkan-profiles
  ==> vulkaninfo
  ERROR: [MoltenVK] Code 0 : VK_ERROR_INCOMPATIBLE_DRIVER: Vulkan is not supported on this device. MoltenVK requires Metal, which is not available on this device.
  WARNING: [Loader Message] Code 0 : terminator_CreateInstance: Received return code -9 from call to vkCreateInstance in ICD /usr/local/share/vulkan/icd.d/../../../lib/libMoltenVK.dylib. Skipping this driver.
  ERROR: [Loader Message] Code 0 : terminator_CreateInstance: Found no drivers!
  Cannot create Vulkan instance.
  This problem is often caused by a faulty installation of the Vulkan driver or attempting to use a GPU that does not support Vulkan.
  ERROR at /tmp/vulkan-tools-20240710-20601-levivl/Vulkan-Tools-1.3.289/vulkaninfo/./vulkaninfo.h:454:vkCreateInstance failed with ERROR_INCOMPATIBLE_DRIVER
  Error: vulkan-profiles: failed
  Error: vulkan-profiles: failed
  An exception occurred within a child process:
    Minitest::Assertion: Expected: 0
    Actual: 1
```

File: Formula/v/vulkan-profiles.rb
Comment (chenrui333) [inline Formula/v/vulkan-profiles.rb:70]: looks like intel macos runner still not support it, revert this for the merge

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**
